kvbc/src/direct_kv_db_adapter.cpp

```diff
@@ -412,10 +413,27 @@ void DBAdapter::addRawBlock(const RawBlock &block, const BlockId &blockId) {
   // when ST runs, blocks arrive in batches in reverse order. we need to keep
   // track on the "Gap" and to close it. Only when it is closed, the last
   // reachable block becomes the same as the last block
-  BlockId lastReachableBlock = getLastReachableBlockId();
-  if (blockId > getLatestBlockId()) setLatestBlock(blockId);
 
-  if (blockId == lastReachableBlock + 1) setLastReachableBlockNum(getLatestBlockId());
+  // old cycle
```

Comment:
nit: did you mean to leave lines 417-427 as a comment?
In my opinion, it should be removed...
If you decide to keep it,  better I'm multiline in one section:
```
  /*
....
  */
```